[PATCH] From ece218405fe4c086017f2a4f2b2b4c86b9790db0 Mon Sep 17

From: Saikrishna Arcot <sarcot@microsoft.com>
Date: Wed, 6 Jul 2022 11:28:16 -0700
 00:00:00 2001 Subject: [PATCH] Make changes to support crosscompilation in
 SONiC

Signed-off-by: Saikrishna Arcot <sarcot@microsoft.com>
---
 debian/frr-doc.info |    1 -
 debian/frr.manpages |   16 ----------------
 debian/rules        |   17 ++++++++++++-----
 3 files changed, 12 insertions(+), 22 deletions(-)
 delete mode 100644 debian/frr-doc.info
 delete mode 100644 debian/frr.manpages

diff --git a/debian/frr-doc.info b/debian/frr-doc.info
deleted file mode 100644
index 1976365e3..000000000
--- a/debian/frr-doc.info
+++ /dev/null
@@ -1 +0,0 @@
-build/doc/user/_build/texinfo/frr.info
diff --git a/debian/frr.manpages b/debian/frr.manpages
deleted file mode 100644
index 5a1b74cfc..000000000
--- a/debian/frr.manpages
+++ /dev/null
@@ -1,16 +0,0 @@
-build/doc/manpages/_build/man/frr-bgpd.8
-build/doc/manpages/_build/man/frr-eigrpd.8
-build/doc/manpages/_build/man/frr-fabricd.8
-build/doc/manpages/_build/man/frr-isisd.8
-build/doc/manpages/_build/man/frr-ldpd.8
-build/doc/manpages/_build/man/frr-nhrpd.8
-build/doc/manpages/_build/man/frr-ospf6d.8
-build/doc/manpages/_build/man/frr-ospfd.8
-build/doc/manpages/_build/man/frr-pimd.8
-build/doc/manpages/_build/man/frr-ripd.8
-build/doc/manpages/_build/man/frr-ripngd.8
-build/doc/manpages/_build/man/frr-watchfrr.8
-build/doc/manpages/_build/man/frr-zebra.8
-build/doc/manpages/_build/man/frr.1
-build/doc/manpages/_build/man/mtracebis.8
-build/doc/manpages/_build/man/vtysh.1
diff --git a/debian/rules b/debian/rules
index 74573c9a0..e0b626fbf 100755
--- a/debian/rules
+++ b/debian/rules
@@ -39,10 +39,17 @@ else
   CONF_GRPC=--enable-grpc
 endif
 
+DEB_BUILD_ARCH ?= $(shell dpkg-architecture -qDEB_BUILD_ARCH)
+DEB_HOST_ARCH ?= $(shell dpkg-architecture -qDEB_HOST_ARCH)
+
+ifneq ($(DEB_BUILD_ARCH),$(DEB_HOST_ARCH))
+export PYTHON=/python_virtualenv/env3/bin/python3
+else
 export PYTHON=python3
+endif
 
 %:
-	dh $@ -Bbuild --with=sphinxdoc
+	dh $@ -Bbuild --with=sphinxdoc --builddirectory=build
 
 override_dh_auto_configure:
 	$(shell dpkg-buildflags --export=sh); \
@@ -80,10 +87,10 @@ override_dh_auto_configure:
 override_dh_auto_install:
 	dh_auto_install
 
-	sed -e '1c #!/usr/bin/python3' -i debian/tmp/usr/lib/frr/frr-reload.py
-	sed -e '1c #!/usr/bin/python3' -i debian/tmp/usr/lib/frr/generate_support_bundle.py
-	sed -e '1c #!/usr/bin/python3' -i debian/tmp/usr/lib/frr/frr_babeltrace.py
-	sed -e '1c #!/usr/bin/python3' -i debian/tmp/usr/lib/frr/ospfclient.py
+	sed -e '1c #!$(shell which $PYTHON)' -i debian/tmp/usr/lib/frr/frr-reload.py
+	sed -e '1c #!$(shell which $PYTHON)' -i debian/tmp/usr/lib/frr/generate_support_bundle.py
+	sed -e '1c #!$(shell which $PYTHON)' -i debian/tmp/usr/lib/frr/frr_babeltrace.py
+	sed -e '1c #!$(shell which $PYTHON)' -i debian/tmp/usr/lib/frr/ospfclient.py
 
 # let dh_systemd_* and dh_installinit do their thing automatically
 	cp build/tools/frr.service debian/frr.service
